File: rb/features.rb
# Europaplus SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module EuropaplusFeatures
  def self.make_feature(name)
    case name
    when "base"
      EuropaplusBaseFeature.new
    when "ratelimit"
      EuropaplusRatelimitFeature.new
    when "retry"
      EuropaplusRetryFeature.new
    when "test"
      EuropaplusTestFeature.new
    when "timeout"
      EuropaplusTimeoutFeature.new
    else
      EuropaplusBaseFeature.new
    end
  end
end

File: py/features.py
# Europaplus SDK feature factory

from feature.base_feature import EuropaplusBaseFeature
from feature.test_feature import EuropaplusTestFeature


def _make_feature(name):
    features = {
        "base": lambda: EuropaplusBaseFeature(),
        "test": lambda: EuropaplusTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
